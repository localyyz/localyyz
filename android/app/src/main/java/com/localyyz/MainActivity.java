package com.localyyz;

import android.content.Intent;
import android.os.Bundle;

import com.facebook.react.ReactActivity;
import com.facebook.react.ReactActivityDelegate;

import javax.annotation.Nullable;

import io.branch.rnbranch.RNBranchModule;

public class MainActivity extends ReactActivity {

    /**
     * Returns the name of the main component registered from JavaScript.
     * This is used to schedule rendering of the component.
     */
    @Override
    protected String getMainComponentName() {
        return "localyyz";
    }

    @Override
    protected void onStart() {
        super.onStart();
        RNBranchModule.initSession(getIntent().getData(), this);
    }

    @Override
    public void onNewIntent(Intent intent) {
        setIntent(intent);
    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        super.onActivityResult(requestCode, resultCode, data);
        MainApplication.getCallbackManager().onActivityResult(requestCode, resultCode, data);
    }

    @Override
    protected ReactActivityDelegate createReactActivityDelegate() {
        return new ReactActivityDelegate(this, getMainComponentName()) {
            @Nullable
            @Override
            protected Bundle getLaunchOptions() {
                Bundle initialProps = new Bundle();
                initialProps.putString("API_URL", BuildConfig.API_URL);
                return initialProps;
            }
        };
    }

}
