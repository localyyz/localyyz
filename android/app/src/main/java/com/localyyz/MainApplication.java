package com.localyyz;

import android.app.Application;

import com.BV.LinearGradient.LinearGradientPackage;
import com.cmcewen.blurview.BlurViewPackage;
import com.facebook.CallbackManager;
import com.facebook.FacebookSdk;
import com.facebook.appevents.AppEventsLogger;
import com.facebook.react.ReactApplication;
import com.geektime.rnonesignalandroid.ReactNativeOneSignalPackage;
import com.idehub.GoogleAnalyticsBridge.GoogleAnalyticsBridgePackage;
import com.brentvatne.react.ReactVideoPackage;
import com.facebook.react.ReactNativeHost;
import com.facebook.react.ReactPackage;
import com.facebook.react.shell.MainReactPackage;
import com.facebook.reactnative.androidsdk.FBSDKPackage;
import com.facebook.soloader.SoLoader;
import com.learnium.RNDeviceInfo.RNDeviceInfo;
import com.microsoft.codepush.react.CodePush;
import com.oblador.vectoricons.VectorIconsPackage;
import com.reactnative.photoview.PhotoViewPackage;

import java.util.Arrays;
import java.util.List;

import ca.jaysoo.extradimensions.ExtraDimensionsPackage;
import io.branch.referral.Branch;
import io.branch.rnbranch.RNBranchPackage;

//import com.reactnativepayments.ReactNativePaymentsPackage;

public class MainApplication extends Application implements ReactApplication {

  private static CallbackManager mCallbackManager = CallbackManager.Factory.create();

  protected static CallbackManager getCallbackManager() {
    return mCallbackManager;
  }

  private final ReactNativeHost mReactNativeHost = new ReactNativeHost(this) {

    @Override
    protected String getJSBundleFile() {
      return CodePush.getJSBundleFile();
    }
    
    @Override
    public boolean getUseDeveloperSupport() {
      return BuildConfig.DEBUG;
    }

    @Override
    protected List<ReactPackage> getPackages() {
      return Arrays.<ReactPackage>asList(
          new MainReactPackage(),
            new ReactNativeOneSignalPackage(),
            new GoogleAnalyticsBridgePackage(),
            new ReactVideoPackage(),
            new ExtraDimensionsPackage(),
              new CodePush(BuildConfig.CODEPUSH_KEY, getApplicationContext(), BuildConfig.DEBUG),
              new RNBranchPackage(),
              //new ReactNativePaymentsPackage(),
              new FBSDKPackage(mCallbackManager),
              new PhotoViewPackage(),
              new VectorIconsPackage(),
              new LinearGradientPackage(),
              new RNDeviceInfo(),
              new BlurViewPackage()
      );
    }

    @Override
    public List<ReactPackage> createAdditionalReactPackages() {
      return getPackages();
    }


    @Override
    protected String getJSMainModuleName() {
      return "index";
    }
  };

  @Override
  public ReactNativeHost getReactNativeHost() {
    return mReactNativeHost;
  }

  @Override
  public void onCreate() {
    super.onCreate();

    FacebookSdk.sdkInitialize(getApplicationContext());
    // If you want to use AppEventsLogger to log events.
    AppEventsLogger.activateApp(this);

    // Branch
    Branch.getAutoInstance(this);

    SoLoader.init(this, /* native exopackage */ false);
  }

}
