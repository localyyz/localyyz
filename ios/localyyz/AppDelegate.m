/**
 * Copyright (c) 2015-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

#import "AppDelegate.h"

//#import <React/RCTBundleURLProvider.h>
#import <React/RCTRootView.h>

// Branch
#import <react-native-branch/RNBranch.h>

// Facebook
#import <FBSDKCoreKit/FBSDKCoreKit.h>
#import <FBSDKLoginKit/FBSDKLoginKit.h>
#import <FBNotifications/FBNotifications.h>
#import <UserNotifications/UserNotifications.h>

// CodePush
#import <CodePush/CodePush.h>

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  NSURL *jsCodeLocation;

  if (RCT_DEBUG == 1) {
    //jsCodeLocation = [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index" fallbackResource:nil];
    jsCodeLocation = [NSURL URLWithString:@"http://localhost:8081/index.bundle?platform=ios&dev=true"];
  } else {
    jsCodeLocation = [CodePush bundleURL];
  }

  NSString *apiUrl = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"API_URL"];
  NSDictionary *props = @{@"API_URL" : apiUrl};
  
  RCTRootView *rootView = [[RCTRootView alloc] initWithBundleURL:jsCodeLocation
                                                      moduleName:@"localyyz"
                                               initialProperties:props
                                                   launchOptions:launchOptions];
  rootView.backgroundColor = [[UIColor alloc] initWithRed:1.0f green:1.0f blue:1.0f alpha:1];
  
  self.window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
  UIViewController *rootViewController = [UIViewController new];
  rootViewController.view = rootView;
  self.window.rootViewController = rootViewController;
  [self.window makeKeyAndVisible];

  // replace RN load view with the launch screen
  UIView* launchScreenView = [[[NSBundle mainBundle] loadNibNamed:@"LaunchScreen" owner:self options:nil] objectAtIndex:0];
  launchScreenView.frame = self.window.bounds;
  rootView.loadingView = launchScreenView;

  // facebook
  [[FBSDKApplicationDelegate sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];

  // branch
  [RNBranch initSessionWithLaunchOptions:launchOptions isReferrable:YES];

  // push notification
  UNUserNotificationCenter *center = [UNUserNotificationCenter currentNotificationCenter];
  center.delegate = self;
  [center requestAuthorizationWithOptions:(UNAuthorizationOptionSound | UNAuthorizationOptionAlert | UNAuthorizationOptionBadge) completionHandler:^(BOOL granted, NSError * _Nullable error) {
    if( !error ) {
      // required to get the app to do anything at all about push notifications
      [[UIApplication sharedApplication] registerForRemoteNotifications];
      NSLog( @"Push registration success.");
    } else {
      NSLog( @"Push registration FAILED" );
      NSLog( @"ERROR: %@ - %@", error.localizedFailureReason, error.localizedDescription );
      NSLog( @"SUGGESTIONS: %@ - %@", error.localizedRecoveryOptions, error.localizedRecoverySuggestion );
    }
  }];

  return YES;
}

- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo fetchCompletionHandler:(nonnull void (^)(UIBackgroundFetchResult))completionHandler {
  FBNotificationsManager *notificationsManager = [FBNotificationsManager sharedManager];
  [notificationsManager presentPushCardForRemoteNotificationPayload:userInfo
                                                 fromViewController:nil
                                                         completion:^(FBNCardViewController * _Nullable viewController, NSError * _Nullable error) {
                                                           if (error) {
                                                             completionHandler(UIBackgroundFetchResultFailed);
                                                           } else {
                                                             completionHandler(UIBackgroundFetchResultNewData);
                                                           }
                                                         }];
}

- (BOOL)application:(UIApplication *)application
            openURL:(NSURL *)url
            options:(NSDictionary<UIApplicationOpenURLOptionsKey,id> *)options {
  
  BOOL handled = [[FBSDKApplicationDelegate sharedInstance] application:application
                                                                openURL:url
                                                      sourceApplication:options[UIApplicationOpenURLOptionsSourceApplicationKey]
                                                             annotation:options[UIApplicationOpenURLOptionsAnnotationKey]
                  ];
  // Add any custom logic here.
  return handled;
}

// track opening and resuming the application attached to the push notification
- (void)application:(UIApplication *)application didReceiveRemoteNotification:(NSDictionary *)userInfo {
  [FBSDKAppEvents logPushNotificationOpen:userInfo];
}

// track action performed that was attached to the push notification
- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forRemoteNotification:(NSDictionary *)userInfo
  completionHandler:(void (^)())completionHandler {
  [FBSDKAppEvents logPushNotificationOpen:userInfo action:identifier];
}

- (BOOL)application:(UIApplication *)application continueUserActivity:(NSUserActivity *)userActivity restorationHandler:(void (^)(NSArray *restorableObjects))restorationHandler {
  return [RNBranch continueUserActivity:userActivity];
}

- (void)application:(UIApplication *)application didRegisterForRemoteNotificationsWithDeviceToken:(NSData *)deviceToken {
  NSString *token = [[deviceToken description] stringByTrimmingCharactersInSet: [NSCharacterSet characterSetWithCharactersInString:@"<>"]];
  token = [token stringByReplacingOccurrencesOfString:@" " withString:@""];
  NSLog(@"content--- %@", token);
  [FBSDKAppEvents setPushNotificationsDeviceToken:deviceToken];
}

// Facebook SDK
- (void)applicationDidBecomeActive:(UIApplication *)application {
  [FBSDKAppEvents activateApp];
}

@end
