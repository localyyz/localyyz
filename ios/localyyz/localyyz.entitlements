<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>aps-environment</key>
	<string>development</string>
	<key>com.apple.developer.associated-domains</key>
	<array>
		<string>applinks:localyyz.app.link</string>
		<string>applinks:localyyz-alternate.app.link</string>
		<string>applinks:localyyz.test-app.link</string>
		<string>applinks:localyyz-alternate.test-app.link</string>
	</array>
	<key>com.apple.developer.in-app-payments</key>
	<array>
		<string>merchant.com.localyyz.toronto</string>
	</array>
</dict>
</plist>
