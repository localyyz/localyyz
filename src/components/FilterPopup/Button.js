import React from "react";
import { View, StyleSheet, TouchableOpacity } from "react-native";
import { Styles, Colours, Sizes } from "localyyz/constants";
import UppercasedText from "../UppercasedText";
import PropTypes from "prop-types";

// third party
import { View as AnimatableView } from "react-native-animatable";
import { withNavigation } from "react-navigation";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import LinearGradient from "react-native-linear-gradient";

import Popup from "./Popup";

export class Button extends React.Component {
  static propTypes = {
    store: PropTypes.object.isRequired,

    text: PropTypes.string,
    isInitialVisible: PropTypes.bool,
    onPress: PropTypes.func
  };

  static defaultProps = {
    text: "Filter / Sort",
    isInitialVisible: false
  };

  componentDidMount() {
    this.props.isInitialVisible ? this.onPress() : null;
  }

  onPress = () => {
    this.props.onPress
      ? this.props.onPress()
      : this.props.navigation.navigate("Modal", {
          component: <Popup {...this.props} />
        });
  };

  render() {
    return (
      <View style={styles.toggle} pointerEvents="box-none">
        <LinearGradient
          colors={[Colours.WhiteTransparent, Colours.Transparent]}
          start={{ x: 0, y: 1 }}
          end={{ x: 0, y: 0 }}
          style={styles.gradient}
          pointerEvents="box-none">
          <TouchableOpacity onPress={this.onPress}>
            <AnimatableView
              animation="fadeIn"
              duration={500}
              delay={1000}
              style={styles.toggleContainer}>
              <MaterialIcon
                name="sort"
                size={Sizes.TinyText}
                color={Colours.Text}/>
              <UppercasedText style={styles.toggleLabel}>
                {this.props.text}
              </UppercasedText>
            </AnimatableView>
          </TouchableOpacity>
        </LinearGradient>
      </View>
    );
  }
}

export default withNavigation(Button);

const styles = StyleSheet.create({
  gradient: {
    alignItems: "center",
    justifyContent: "flex-end",
    height: Sizes.Height / 6
  },

  toggleContainer: {
    ...Styles.Horizontal,
    margin: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 2,
    paddingHorizontal: Sizes.InnerFrame,
    borderRadius: Sizes.InnerFrame,
    backgroundColor: Colours.WhiteTransparent,
    shadowColor: Colours.DarkTransparent,
    shadowOffset: { width: 0, height: 0 },
    shadowRadius: 5,
    shadowOpacity: 0.1,
    alignItems: "center",
    justifyContent: "center"
  },

  toggleLabel: {
    ...Styles.Text,
    ...Styles.SmallText,
    ...Styles.Medium,
    marginLeft: Sizes.InnerFrame / 2
  }
});
