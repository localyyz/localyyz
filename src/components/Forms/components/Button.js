import React from "react";
import {
  View,
  StyleSheet,
  TouchableOpacity,
  TouchableWithoutFeedback,
  Text
} from "react-native";

// third party
import PropTypes from "prop-types";
import { inject, observer } from "mobx-react/native";

// custom
import { Styles, Sizes, Colours } from "localyyz/constants";

@inject(stores => ({
  isComplete: stores.formStore && stores.formStore.isComplete
}))
@observer
export default class Button extends React.Component {
  static propTypes = {
    children: PropTypes.string.isRequired,
    onPress: PropTypes.func,

    // mobx injected
    isComplete: PropTypes.bool
  };

  static defaultProps = {
    onPress: () => {},
    isComplete: false
  };

  get touchableType() {
    return this.isDisabled ? TouchableWithoutFeedback : TouchableOpacity;
  }

  get isDisabled() {
    return (
      !this.props.isEnabled
      && (!!this.props.isDisabled || !this.props.isComplete)
    );
  }

  render() {
    return (
      <View style={styles.container}>
        <this.touchableType
          onPress={!this.isDisabled ? this.props.onPress : null}>
          <View
            style={[
              styles.box,
              this.isDisabled && styles.disabled,
              this.props.color && { backgroundColor: this.props.color }
            ]}>
            <Text
              style={[
                styles.label,
                this.isDisabled && styles.disabledLabel,
                this.props.labelColor && { color: this.props.labelColor }
              ]}>
              {this.props.children}
            </Text>
          </View>
        </this.touchableType>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    marginTop: Sizes.InnerFrame,
    alignItems: "center"
  },

  box: {
    ...Styles.RoundedButton
  },

  disabled: {
    backgroundColor: Colours.DisabledButton
  },

  label: {
    ...Styles.Text,
    ...Styles.Emphasized,
    ...Styles.Alternate
  },

  disabledLabel: {
    ...Styles.Subdued
  }
});
