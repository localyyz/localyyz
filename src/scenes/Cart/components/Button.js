import React from "react";
import {
  View,
  StyleSheet,
  Text,
  TouchableOpacity,
  ActivityIndicator
} from "react-native";

// third party
import { observer, inject } from "mobx-react/native";
import * as Animatable from "react-native-animatable";
import FontAwesomeIcon from "react-native-vector-icons/FontAwesome";

// custom
import { Colours, Sizes, Styles } from "localyyz/constants";

@inject(stores => ({
  isReady:
    stores && stores.cartUiStore ? stores.cartUiStore.nextSceneIsReady : true
}))
@observer
export default class Button extends React.Component {
  get containerComponent() {
    return this.props.onPress && this.props.isReady ? TouchableOpacity : View;
  }

  get ready() {
    return (
      <View style={styles.content}>
        {this.props.icon ? (
          <FontAwesomeIcon
            name={this.props.icon}
            size={Sizes.Text}
            color={Colours.AlternateText}
            style={styles.icon}/>
        ) : null}
        <Text style={styles.label}>{this.props.children}</Text>
      </View>
    );
  }

  get loading() {
    return (
      <View style={styles.content}>
        <ActivityIndicator size="small" color={Colours.AlternateText} />
      </View>
    );
  }

  render() {
    return (
      <this.containerComponent onPress={this.props.onPress}>
        <Animatable.View
          animation="fadeIn"
          duration={500}
          style={[
            styles.container,
            this.props.backgroundColour && {
              backgroundColor: this.props.backgroundColour
            },
            this.props.style
          ]}>
          {this.props.isReady ? this.ready : this.loading}
        </Animatable.View>
      </this.containerComponent>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    ...Styles.RoundedButton,
    backgroundColor: Colours.MenuBackground
  },

  label: {
    ...Styles.Text,
    ...Styles.Emphasized,
    ...Styles.Alternate
  },

  icon: {
    marginRight: Sizes.InnerFrame / 2
  },

  content: {
    ...Styles.Horizontal
  }
});
