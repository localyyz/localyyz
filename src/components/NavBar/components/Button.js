import React from "react";
import { View, StyleSheet, TouchableOpacity, Text } from "react-native";
import { Colours, Sizes, Styles } from "localyyz/constants";
import PropTypes from "prop-types";

// custom
import { SloppyView } from "localyyz/components";

// third party
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import MaterialCommunityIcon from "react-native-vector-icons/MaterialCommunityIcons";
import EntypoIcon from "react-native-vector-icons/Entypo";
import FontAwesomeIcon from "react-native-vector-icons/FontAwesome";

export default class Button extends React.Component {
  static propTypes = {
    onPress: PropTypes.func.isRequired,
    // button id to sync group activation and deactivation
    id: PropTypes.string.isRequired,
    activeButton: PropTypes.string.isRequired,

    icon: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    badge: PropTypes.string,

    iconType: PropTypes.string
  };

  static defaultProps = {
    badge: null
  };

  static getDerivedStateFromProps(props) {
    return {
      isActive:
        props.activeButton == "cart" ? props.activeButton == props.id : true
    };
  }

  constructor(props) {
    super(props);
    this.state = {
      isActive: props.activeButton == props.id
    };
  }

  get buttonColor() {
    return this.props.activeButton === "deals"
      ? Colours.AlternateText
      : this.state.isActive ? Colours.Text : Colours.SubduedText;
  }

  get icon() {
    switch (this.props.iconType) {
      case "entypo":
        return EntypoIcon;
      case "fontAwesome":
        return FontAwesomeIcon;
      case "material":
        return MaterialIcon;
      default:
        return MaterialCommunityIcon;
    }
  }

  onPress = () => {
    this.setState({ isActive: true });

    // propagate
    this.props.onPress();
  };

  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress}>
        <View key={this.props.label} style={styles.container}>
          <this.icon
            name={this.props.icon}
            size={Sizes.IconButton}
            color={this.buttonColor}/>
          <Text style={[styles.buttonLabel, { color: this.buttonColor }]}>
            {this.props.label}
          </Text>
          {this.props.badge ? (
            <View style={styles.badgeContainer}>
              <View style={styles.badge}>
                <Text style={styles.badgeLabel}>{this.props.badge}</Text>
              </View>
            </View>
          ) : null}
        </View>
      </TouchableOpacity>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "space-between",
    paddingHorizontal: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 4
  },

  buttonLabel: {
    ...Styles.Text,
    ...Styles.TabBarText,
    height: Sizes.InnerFrame
  },

  badgeContainer: {
    position: "absolute",
    alignItems: "flex-end",
    justifyContent: "flex-end",
    left: 0,
    right: Sizes.InnerFrame / 2,
    bottom: Sizes.InnerFrame
  },

  badge: {
    backgroundColor: Colours.Secondary,
    borderRadius: Sizes.InnerFrame,
    paddingVertical: Sizes.InnerFrame / 6,
    paddingHorizontal: Sizes.InnerFrame / 3
  },

  badgeLabel: {
    ...Styles.Text,
    ...Styles.TabBarText,
    ...Styles.Emphasized,
    ...Styles.Alternate
  }
});
