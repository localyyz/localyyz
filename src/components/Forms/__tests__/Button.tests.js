import React from "react";

// custom
import { renderWithLayout } from "localyyz/tests";

// local
import Button from "../components/Button";

// constants
const NAME = "Button test - form";
const PROPS = {
  children: "sample children",
  onPress: jest.fn(),
  color: "green",
  labelColor: "red"
};
const Component = Button.wrappedComponent;

describe(NAME, () => {
  it(`${NAME}: should render properly`, () => {
    let c = renderWithLayout(<Component {...PROPS} />).getInstance();
    expect(c).not.toBeUndefined();
  });

  it(`${NAME}: should render to snapshot: isDisabled() => false`, () => {
    PROPS.isEnabled = true;
    PROPS.isDisabled = true;
    PROPS.isComplete = true;
    expect(
      renderWithLayout(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });

  it(`${NAME}: should render to snapshot: isDisabled() => true`, () => {
    PROPS.isEnabled = false;
    PROPS.isDisabled = true;
    PROPS.isComplete = true;
    expect(
      renderWithLayout(<Component {...PROPS} />).toJSON()
    ).toMatchSnapshot();
  });
});
