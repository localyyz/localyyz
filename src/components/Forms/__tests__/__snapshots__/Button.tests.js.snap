// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test - form Button test - form: should render to snapshot: isDisabled() => false 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 15,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#9013FE",
            "borderRadius": 15,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 30,
            "paddingVertical": 7.5,
          },
          false,
          Object {
            "backgroundColor": "green",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(255, 255, 255, 0)",
              "borderBottomColor": "#FFFFFF",
              "color": "#FFFFFF",
              "fontSize": 16,
              "fontWeight": "600",
            },
            false,
            Object {
              "color": "red",
            },
          ]
        }
      >
        sample children
      </Text>
    </View>
  </View>
</View>
`;

exports[`Button test - form Button test - form: should render to snapshot: isDisabled() => true 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "marginTop": 15,
    }
  }
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#9013FE",
          "borderRadius": 15,
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 30,
          "paddingVertical": 7.5,
        },
        Object {
          "backgroundColor": "rgba(255, 255, 255, 0.5)",
        },
        Object {
          "backgroundColor": "green",
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "backgroundColor": "rgba(255, 255, 255, 0)",
            "borderBottomColor": "#FFFFFF",
            "color": "#FFFFFF",
            "fontSize": 16,
            "fontWeight": "600",
          },
          Object {
            "color": "#999999",
          },
          Object {
            "color": "red",
          },
        ]
      }
    >
      sample children
    </Text>
  </View>
</View>
`;
